go.mod, go.sum, freezer_remote_s3_test.go: sketch of what test harnes…

zcstarr opened the pull request:
…s might look like

zcstarr committed:
go.mod, go.sum, freezer_remote_s3_test.go: sketch of what test harness might look like

zcstarr commented:
For ease of viewing we have a bug with two of the tests from the blockchain_remote core-geth tests. You should be able to repro it via .

go test . This assumes you have aws credentials setup. There is a similar PR in the storj repo same issues.

meowsbits committed:
main: use testName as bucket name parameter

Using a random int for a bucket name
isn't going to work. Then we have which
test data who-knows-where.

This uses the same bucket for each test.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
main: refactor: simplify test commmand args assignment

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
main: print error value rather than wrap

wrapping isnt compatible with the
fancy error type that s3 lib might
return us

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
main: refactor quite a few things

- fix bucket name, see link in comments
- allow externally defined s3 credentials
- cleanup temporary ipc paths for each test
- use test.Run to idiomatically break tests into
subtests

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: bump core-geth dep

Signed-off-by: meows <b5c6@protonmail.com>

zcstarr committed:
main.go: fix close to NOP vs. stop rpc server

meowsbits committed:
Revert "main.go: fix close to NOP vs. stop rpc server"

This reverts commit 7b06c96b71e8ba7fc786268cc8ddbc494f3042e5.

meowsbits committed:
main: fix quit patterns for client/server at test

Each test instantiates a new freezer server, and is
responsible for shutting that app down when the
test has finished.

The Close() method handled by the server does not
indicate that the server should close, but that
that client is Close-ing (eg. client quitting).

The S3 freezer server logic doesn't do much on Close,
if anything, so this call is simply read and logged.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
main: skip live-s3 intergration tests if credentials fail

Valid credentials for S3 are required for live
resource integration testing.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits commented:
Failing tests cited here https://github.com/etclabscore/ancient-store-s3/pull/1#issuecomment-666947597 appear to be resolved, and were issues with the integration test(s) itself, rather than the store implementation.

meowsbits committed:
Revert "main: skip live-s3 intergration tests if credentials fail"

This reverts commit 6068b7693b7ec49ae9f217005b2c60d7a4c39b92.

meowsbits committed:
main: install configurable s3 bucket namespacing

See comments; bucket namespacing will be necessary
since s3 bucket names must be globally unique.

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
.github/workflows/go.yml: install s3 credentials for github action

Adds repository secrets and uses them in
configuring the environment to be able to
run live-resource tests.

Signed-off-by: meows <b5c6@protonmail.com>
